api/nodes/views.py

```diff
@@ -2961,3 +2962,27 @@ def get_parser_context(self, http_request):
         res = super(LinkedNodesList, self).get_parser_context(http_request)
         res['is_relationship'] = True
         return res
+
+class UserPublicFiles(NodeFilesList, UserMixin):
```

Comment:
If the api endpoint remains ```/users/<user-id>/public_files/```, then this view should be moved to ```api/users/views.py```. Additionally, this endpoint needs tests for retriving, creating, updating, and deleting public files.